lib/dry/validation/rule.rb

```diff
@@ -82,7 +82,7 @@ def each(*macros, &block)
         @keys = []
 
         @block = proc do
-          (values[root] || []).each_with_index do |_, idx|
+          ((values[root] if values[root].is_a?(Array)) || []).each_with_index do |_, idx|
```

Comment:
Ideally we should not reach this code if params validation does not pass, but I am not sure how do that yet.